MapboxNavigation/ImageCache.swift

```diff
@@ -99,67 +55,29 @@ class ImageCache: BimodalImageCache {
         memoryCache.removeAllObjects()
     }
 
-    func clearDisk(completion: NoArgBlock?) {
-        guard let fileManager = fileManager else {
-            return
-        }
-        let cacheURL = self.diskCacheURL
-        self.diskAccessQueue.async {
-            do {
-                try fileManager.removeItem(at: cacheURL)
-            } catch {
-                NSLog("================> Failed to remove cache dir: \(cacheURL)")
-            }
-
-            self.createCacheDirIfNeeded(cacheURL, fileManager: fileManager)
+    func clearDisk(completion: CompletionHandler?) {
+        fileCache.clearDisk(completion: completion)
+    }
 
-            if let completion = completion {
-                DispatchQueue.main.async {
-                    completion()
-                }
-            }
-        }
+    private func cacheCostForImage(_ image: UIImage) -> Int {
+        return Int(image.size.height * image.size.width * image.scale * image.scale);
```

Comment:
Should scale be multiplied here twice?

Reply:
yeah, it reads funny but it's once for `x`, once for `y`

we could also just delete this and let the underlying memory cache figure it out

Reply:
I like this quite a bit, but perhaps we could make it a little more self-documenting? It might not read so tricky that way. 

Also, I feel like the name could be a little more [swift-y](https://swift.org/documentation/api-design-guidelines/#promote-clear-usage). The prefix `cache` is redundant in the function because the class is `ImageCache`, and the naming could be a little more [concise](https://swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage), especially considering the `private` nature of the function.
```swift
    private func cost(for image: UIImage) -> Int {
       let scale = image.scale

       //calculate horizontal cost, in pixels
       let horizontalCost = image.size.width * scale

       //calculate vertical cost, in pixels
       let verticalCost = image.size.height * scale

       return Int(horizontalCost * verticalCost)
   }
```

The usage-at-callsite is quite a bit [cleaner](https://swift.org/documentation/api-design-guidelines/#fundamentals) this way, being used as `cost(for: shield)` instead of `cacheCostForImage(shield)`.